src/components/account/Profile/index.tsx

```diff
@@ -1,11 +1,232 @@
+import GithubIcon from '../../general/Icon/GithubIcon'
+import { ShareModalContent } from '../Share/Share'
+import { AnimatePresence, m } from 'framer-motion'
+import type { ENSRecords, ResolvedENS } from 'get-ens'
+// eslint-disable-next-line no-duplicate-imports
+import { getENS } from 'get-ens'
+import Spinner from '../../general/Spinner'
+import EmailIcon from '../../general/Icon/EmailIcon'
+import RedditIcon from '../../general/Icon/RedditIcon'
+
+const ProfileLink = ({
+  title,
+  url,
+  icon
+}: {
+  title: string
+  url: string
+  icon: React.ReactNode
+}) => {
+  return (
+    <a href={url} target="_blank" className="Profile__Container__socials--item">
+      {icon}
+      <span>{title}</span>
+    </a>
+  )
+}
+
+interface ModifiedResolvedENS extends ResolvedENS {
+  records: ENSRecords
+}
+
+const ProfileModalContent: React.FC<{
+  handleShareClick: () => void
+}> = ({ handleShareClick }) => {
+  const provider = useProvider()
+  const { userPubkey: address } = useContext(W3iContext)
+  const locallyStoredData = localStorage.getItem('ens-records')
+  const [resolvedRecords, setResolvedRecords] = useState<ENSRecords | undefined>(
+    locallyStoredData ? JSON.parse(locallyStoredData) : undefined
+  )
+  const addressOrEnsDomain = address as `0x${string}` | undefined
+  const { data: ensName } = useEnsName({ address: addressOrEnsDomain })
+  const { data: ensAvatar } = useEnsAvatar({ address: addressOrEnsDomain })
+
+  useEffect(() => {
+    const getData = async () => {
+      const data = (await getENS({ provider })(ensName ?? '')) as ModifiedResolvedENS
+      setResolvedRecords(data.records)
+      localStorage.setItem('ens-records', JSON.stringify(data.records))
+    }
+    if (!locallyStoredData) {
+      getData()
+    }
+  }, [])
+
+  return (
+    <m.div
+      initial={{
+        opacity: 0,
+        x: '-100%'
+      }}
+      animate={{
+        opacity: 1.01,
+        x: '0%'
+      }}
+      exit={{
+        opacity: 0,
+        x: '-100%',
+        transition: { duration: 0.1 }
+      }}
+      transition={{ type: 'spring', duration: 0.4, bounce: 0.15 }}
+      className="Profile"
+      style={{
+        position: 'static',
+        ...(address ? generateAvatarColors(address) : {})
+      }}
+    >
+      <div className="Profile__TitleBar">
+        <div className="Profile__TitleBar__buttons">
+          <button onClick={handleShareClick}>
+            <ShareIcon />
+          </button>
+          <button
+            onClick={() => {
+              profileModalService.toggleModal()
+              localStorage.removeItem('ens-records')
+            }}
+          >
+            <CrossIcon />
+          </button>
+        </div>
+        <div className="Profile__TitleBar__background__container">
+          <div className="Profile__TitleBar__background" />
+        </div>
+        <div className="Profile__icon">{ensAvatar && <img src={ensAvatar} alt="ENS Avatar" />}</div>
```

Comment:
Couldn't we use `Avatar` component for this? As it'd generate unique background gradients for each addres?

Reply:
Right!! I didnt realise we had that, mb